### DISCUSSION TITLE
Prevent Duplicate file upload

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#14370] [Archive]
How do you prevent a duplicate file upload,
if the file is saved on disk with a hash?

**Comment by benhaynes** [comment]
We don't currently check for that. I believe we had a file hash previously, but that was difficult on larger files. I think there is another discussion for this on GitHub, but would love to have that hash and warn users about duplicate files on upload.

**Reply by AlexVanK** [reply]
Yeah, this is a feature that is very much required at least for our use case...
I'm trying to upload 2700 files, and when the upload fails, I cannot just re-attempt a drag&drop of the same files chunk, because I would be duplicating 100s of files over and over again. 

On a side note, It would be useful no notify the user as to which file has failed the upload, instead of seeing the general error "Token has expired".

Anyway, I'm working on a large form system that asks the users to upload many pieces of attachment files. These files feed a public library and need to be maintained over time. With a hash saving feature on disk, it seems to be impossible to "replace" those files, without looking up the hash first, and also maintenance and cleanup of the files collection from the backend can become a challenge...
